### DISCUSSION TITLE
phc2sys issue

### DISCUSSION THREAD

**Discussion by FaisalJAbushanab** [srsran/srsRAN_Project#669] [General Help]
Hi all, 

We are trying to do a split 7.2 setup using srs gnb and different RUs that were tested by srs (i.e. Benetel RAN550). All info about the system is at the end.

one thing we notice is in the time sync process, when we run the srsRAN gNB, the phc2sys offset shoots up and becomes unstable. 

To explain better, see below:

1. We connect our GM to the PC that contains the gNB, configure IP addresses and everything
2. Wait for GM to get GNSS lock, then enable PTP for the PC.
3. on the PC, run ptp4l and phc2sys **(ptp-RAN550.cfg will be shared below)**:
    a. `sudo ./ptp4l -2 -i enp1s0f2np2 -s -f ./configs/ptp-RAN550.cfg -m`
    
![image](https://github.com/srsran/srsRAN_Project/assets/33567969/f7eb3cf6-e234-40dd-aa55-166818a8c7c2)
    b. `sudo ./phc2sys -s enp1s0f2np2 -w -m -R 8 -f ./configs/ptp-RAN550.cfg`
     
![image](https://github.com/srsran/srsRAN_Project/assets/33567969/de85fcab-b4b7-434d-8433-2ffb8ecdab17)

**So as you can see, rms is almost 0 and is stable, offset is almost 0 and is stable.** 

However, **as soon as we run gNB, this happens to phc2sys output:**
![image](https://github.com/srsran/srsRAN_Project/assets/33567969/8c04d327-7b99-4c2f-b663-39197ba233b5)
Notice how it jumps from 2 to 59 then back to -4 and so on. It seems unstable to me.

It doesn’t matter if the RU on the other side is connected or not, the type of RU doesn’t matter as well, it seems that just running the gNB executable causes this.  

And, we’re facing late packets as reported by gNB logs, so we’re thinking this might be the issue. Did anyone face this? Any clue why the gNB causes this issue? Is it even an issue, cause I can’t find other reasons for the late frames we’re facing.

**Info about the system:**

CPU: Intel i9-13900

RAM: 128 GB

NIC: Intel E810-XXVDA4T

OS: Ubuntu 22.04

Kernel: tried both Real-time 5.15 kernel and the generic kernel, no difference.

the ptp conf file:
[ptp-RAN550.txt](https://github.com/user-attachments/files/15803593/ptp-RAN550.txt)

**Comment by andrepuschmann** [comment]
Hey @FaisalJAbushanab - what you see is expected as `phc2sys` is not isolated in your setup and has to share all resources, e.g. CPU/network stack with the gNB. Also C/U plane and S plane go over the same network interface I guess. Good news is that your values are still low so I don't expect any major issues with the sync. But please not that this is not a RAN issue but more a system configuration topic.
